barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_verifier.cpp

```diff
@@ -190,8 +191,13 @@ bool ProtoGalaxyVerifier_<VerifierInstances>::verify_folding_proof(const std::ve
     auto deltas = compute_round_challenge_pows(accumulator->log_instance_size, delta);
 
     std::vector<FF> perturbator_coeffs(accumulator->log_instance_size + 1);
-    for (size_t idx = 0; idx <= accumulator->log_instance_size; idx++) {
-        perturbator_coeffs[idx] = transcript->template receive_from_prover<FF>("perturbator_" + std::to_string(idx));
+    if (accumulator->is_accumulator) {
+        for (size_t idx = 0; idx <= accumulator->log_instance_size; idx++) {
+            perturbator_coeffs[idx] =
+                transcript->template receive_from_prover<FF>("perturbator_" + std::to_string(idx));
+        }
+    } else {
+        std::fill(perturbator_coeffs.begin(), perturbator_coeffs.end(), 0);
```

Comment:
you can just initialise with 0 a couple of lines up where vector is allocated memory, no need for `std::fill`